backend/backend/services/business_logic.py

```diff
@@ -2,6 +2,44 @@
 
 from utils.logger import logger
 
+import json
+
+
+async def get_test_result(context: Context, subject: str):
+    logger.info(f"Getting test result for subject: {subject}")
+
+    adjacent_chain = context.chains.get_chains()['adjacent_queries_chain']
+    search_tool = context.tools.get_search_tool()
+    summary_chain = context.chains.get_chains()['summary_chain']
+
+    knowledge = []
+    for iter in range(2):
+        queries = await adjacent_chain.ainvoke(subject, 4,
+                                               knowledge="\n".join(knowledge))
+        responses = []
+        for query in queries:
+            artifact = await search_tool.ainvoke_tool_call_artifact(query)
+            response = {"query": query, "answer": artifact["answer"]}
+
+            if False:
+                summary = await summary_chain.ainvoke(query,
+                                                      "\n".join(result["raw_content"]
+                                                                for result in artifact["results"]
+                                                                if result["raw_content"]))
+                response["summary"] = summary
+
+            responses.append(response)
+            knowledge.extend([f"Query: {response["query"]}",
+                              f"Answer: {response["answer"]}",
+                              "========"])
```

Comment:
There is a syntax error in the logging statement on line 34. The closing brace `}` for the `json.dumps` call is misplaced and will cause an error.